### ISSUE TITLE
with A Triangular A_mul_B! overwrites B but Ac_mul_B! doesn't

### ISSUE THREAD

**Issue by dmbates** [JuliaLang/LinearAlgebra.jl#118]
(I meant to write "but Ac_mul_B! doesn't".)

```
julia> B = fill(5.,(1,4))
1x4 Array{Float64,2}:
 5.0  5.0  5.0  5.0

julia> A = Triangular(fill(0.723,(1,1)),:L,false)
1x1 Triangular{Float64}:
 0.723

julia> A_mul_B!(A,B)
1x4 Array{Float64,2}:
 3.615  3.615  3.615  3.615

julia> B
1x4 Array{Float64,2}:
 3.615  3.615  3.615  3.615

julia> B = fill(5.,(1,4))
1x4 Array{Float64,2}:
 5.0  5.0  5.0  5.0

julia> Ac_mul_B!(A,B)
1x4 Array{Float64,2}:
 3.615  3.615  3.615  3.615

julia> B
1x4 Array{Float64,2}:
 5.0  5.0  5.0  5.0

julia> versioninfo()
Julia Version 0.3.0-prerelease+3341
Commit c34e492* (2014-05-30 14:23 UTC)
Platform Info:
  System: Linux (x86_64-pc-linux-gnu)
  CPU: AMD Athlon(tm) II X4 635 Processor
  WORD_SIZE: 64
  BLAS: libopenblas (NO_LAPACK NO_LAPACKE DYNAMIC_ARCH NO_AFFINITY)
  LAPACK: liblapack
  LIBM: libopenlibm
```

**Comment by dmbates** [comment]
It seems that the problem is that the `A_mul_B!` method calls `BLAS.trmm!` but `Ac_mul_B!` calls `BLAS.trmm` (w/o the '!').  

@andreasnoackjensen Is this intentional?

**Comment by andreasnoack** [comment]
Fixed by 547facf2c1755153743ff406a54fb0238464696c
